### ISSUE TITLE
Regression in 0.0.74: classmethod ParamSpec forwarding expects the entire parameter list for each argument

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#4383]
A classmethod whose `cls` parameter is annotated as `Callable[P, ...]` incorrectly rejects valid constructor arguments in ty 0.0.74. The same calls are accepted in 0.0.73.

### Reproducer

```python
from typing import Callable, ParamSpec

P = ParamSpec("P")


class Capybara:
    def __init__(self, widgets: str, n_spam: int, *, out: str | None = None) -> None:
        self.widgets = widgets

    @classmethod
    def apply(cls: Callable[P, "Capybara"], *args: P.args, **kwargs: P.kwargs) -> "Capybara":
        return cls(*args, **kwargs)()

    def __call__(self) -> "Capybara":
        return self


first_positive = Capybara("widgets", n_spam=3)
second_positive = Capybara.apply("widgets", n_spam=3)
```

### Expected behavior

Accept both positive calls.

### Actual behavior

- **0.0.73:** Accepted.
- **0.0.74:** `second_positive` is rejected.

The new diagnostics say:

```text
forward_positive = Sampling.apply("logits", n_vocab=3)
                                  ^^^^^^^^
Expected `(logits: str, n_vocab: int, *, out: str | None = None)`,
found `Literal["logits"]`
```

The `n_vocab=3` argument similarly expects the entire parameter list instead of `int`.

This appears to be incorrect specialization of `P.args` / `P.kwargs` when binding the classmethod. The direct constructor call remains accepted.

### Related issue

#4369 also concerns classmethod/ParamSpec binding, but its example involves `Self` and protocol matching. This reproducer uses neither.

**Comment by carljm** [comment]
Thanks! This blames to my PR https://github.com/astral-sh/ruff/pull/27922, will fix.
